src/Entity/TaskExecutionRepository.php

```diff
@@ -92,6 +92,22 @@ public function findPending(TaskInterface $task)
     /**
      * {@inheritdoc}
      */
+    public function findByUuid($uuid)
```

Comment:
Same opinion as @wachterjohannes. This way it will be easier to optimize if there come up performance problem at some time.